### PR TITLE
edgectl: Fix Mac outbound and add an outbound smoke test

### PR DESCRIPTION
**Pull request by ark3**
Trying #167 again using direct calls to the `edgectl` binary. This includes a fix for MacOS (using `sudo` instead of `su`) that shouldn't have to be a special case, i.e. the same code _should_ work on Linux but does not (at least for me).

Note that Mac CI is b0rked, even for master. A commit that passed CI ~11 days ago now fails consistently. The Mac CI environment itself has changed in a way that appears to be incompatible with current Teleproxy.

### CODE REVIEW

**Comment by LukeShu** [comment]
LGTM, but I'm not clicking "Approve" because I know @rhs was pretty opposed to calling out to Make when it came up over at https://github.com/datawire/teleproxy/pull/168#discussion_r317693372 so I'm not really comfortable clicking "Approve" on this without his OK.

File: pkg/dtest/lock.go [hunk 1]
```diff
@@ -62,5 +64,6 @@ func WithNamedMachineLock(name string, body func()) {
 		file.Close()
 	}()
 
+	fmt.Printf("Acquiring machine lock %q took %.2f seconds\n", name, time.Since(lockAcquireStart).Seconds())
```
Comment (LukeShu) [inline pkg/dtest/lock.go]: Did you mean to leave this in?

Comment (ark3) [reply pkg/dtest/lock.go]: This `Printf` and the associated `grep` pipeline in the CI YAML are in the last commit so they can be dropped easily. I'd prefer to leave them in for now, until we improve the lock situation (e.g., via #179).

**Comment by ark3** [comment]
Once it becomes possible to call Teleproxy as a function (in a Goroutine) instead of as a subprocess, I should be able to rejigger things so that building the executable won't be necessary. I don't think it makes sense to hold up having smoke tests for that. How about I file an issue about this ("Edge Control smoke tests should not build and use the `edgectl` binary") once we land this PR?

**Pull request merged**